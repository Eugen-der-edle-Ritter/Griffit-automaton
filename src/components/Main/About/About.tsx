import React, { FC } from 'react';

import styled from '@emotion/styled';

export const About: FC = () => {
  return (
    <>
      <Title>About cellular automatons</Title>
      <Description>
        A cyclic cellular automaton is a kind of cellular automaton rule
        developed by David Griffeath and studied by several other cellular
        automaton researchers. In this system, each cell remains unchanged until
        some neighboring cell has a modular value exactly one unit larger than
        that of the cell itself, at which point it copies its neighbor's value.
        One-dimensional cyclic cellular automata can be interpreted as systems
        of interacting particles, while cyclic cellular automata in higher
        dimensions exhibit complex spiraling behavior.
      </Description>
    </>
  );
};

const Title = styled.h1`
  margin: 3rem;
  text-align: center;
  font-size: 2.5rem;
`;
const Description = styled.div`
  max-width: 70vw;
  margin: 0 auto;
  font-size: 1.5rem;
  font-style: italic;
`;
