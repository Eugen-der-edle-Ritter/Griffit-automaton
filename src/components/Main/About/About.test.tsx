import React from 'react';
import { render } from '@testing-library/react';

import { About } from './About';

it('About section renders correctly', () => {
  const { asFragment } = render(<About />);

  expect(asFragment()).toMatchSnapshot();
});
