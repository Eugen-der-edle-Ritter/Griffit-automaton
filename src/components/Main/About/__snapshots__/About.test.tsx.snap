// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About section renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="css-zuhk1i"
  >
    About cellular automatons
  </h1>
  <div
    class="css-bzg0kf"
  >
    A cyclic cellular automaton is a kind of cellular automaton rule developed by David Griffeath and studied by several other cellular automaton researchers. In this system, each cell remains unchanged until some neighboring cell has a modular value exactly one unit larger than that of the cell itself, at which point it copies its neighbor's value. One-dimensional cyclic cellular automata can be interpreted as systems of interacting particles, while cyclic cellular automata in higher dimensions exhibit complex spiraling behavior.
  </div>
</DocumentFragment>
`;
